code/messaging/src/phone/java/com/adobe/marketing/mobile/MessageRequiredFieldMissingException.java

```diff
@@ -0,0 +1,27 @@
+/*
+  Copyright 2021 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+package com.adobe.marketing.mobile;
+
+/**
+ * Exception indicating that a required field is missing from the payload definition for a {@code InAppNotification} object.
+ */
+class MessageRequiredFieldMissingException extends Exception {
+
+    /**
+     * Constructor.
+     *
+     * @param message {@link String} containing the message for the new exception
+     */
+    protected MessageRequiredFieldMissingException(final String message) {
```

Comment:
If we don't have to create child class of `MessageRequiredFieldMissingException` we can remove `protected` from constructor signature and make the class final.